lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -839,4 +839,19 @@ void DAP::SetFrameFormat(llvm::StringRef format) {
   }
 }
 
+void DAP::SetThreadFormat(llvm::StringRef format) {
+  if (format.empty())
+    return;
+  lldb::SBError error;
+  g_dap.thread_format = lldb::SBFormat(format.data(), error);
```

Comment:
if `format` isn't null terminated this can cause problems. You will want to use the following to ensure this is NULL terminated:
```
g_dap.thread_format = lldb::SBFormat(format.str().c_str(), error);
```
If the same thing was being done for the `g_dap.frame_format` lets fix that too.

Reply:
good catch. I sometimes forget that StringRef can contain anything